libc/src/string/memory_utils/aie/inline_memcpy.h

```diff
@@ -0,0 +1,34 @@
+//===-- Memcpy implementation for AIE----------------------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2023-2024 Advanced Micro Devices, Inc. or its affiliates
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LIBC_SRC_STRING_MEMORY_UTILS_AIE_INLINE_MEMCPY_H
+#define LIBC_SRC_STRING_MEMORY_UTILS_AIE_INLINE_MEMCPY_H
+
+#include "src/__support/macros/attributes.h"               // LIBC_INLINE
+#include "src/__support/macros/properties/architectures.h" // LIBC_TARGET_ARCH_IS_AIE2
+#include "src/string/memory_utils/generic/aligned_access.h"
+#include "src/string/memory_utils/utils.h" // Ptr, CPtr
+
+#include <stddef.h> // size_t
+
+namespace LIBC_NAMESPACE {
+
+[[maybe_unused]] LIBC_INLINE void
+inline_memcpy_aie2(Ptr __restrict dst, CPtr __restrict src, size_t count) {
```

Comment:
Were you able to test this change?

Reply:
Hi, not through benchmarks. I'm jut linked and inspected the code. However, for now, I dropped this library part to work as a follow-up improvement.